lib/dai-plugin-mcd/src/CdpTypeService.js

```diff
@@ -0,0 +1,45 @@
+import Maker from '@makerdao/dai';
+import { WAD, RAY } from '@makerdao/dai/utils/constants';
+import BigNumber from 'bignumber.js';
+import CdpType from './CdpType';
+
+export default class CdpTypeService extends Maker.PublicService {
+  constructor(name = 'cdpType') {
+    super(name, ['smartContract']);
+  }
+
+  initialize(settings = {}) {
+    if (settings.cdpTypes) {
+      this._cdpTypes = settings.cdpTypes.map(ilk =>
+        this.getCdpTypeObject(ilk.currency)
+      );
+    }
+  }
+
+  // System-Wide Data ----------------------------------------------
+
+  async getAnnualSavingsRate() {
+    const repo = await this._dripContract().repo();
+    const repoBigNumber = new BigNumber(repo.toString()).dividedBy(RAY);
+    const secondsPerYear = 60 * 60 * 24 * 365;
+    BigNumber.config({ POW_PRECISION: 100 });
+    return repoBigNumber.pow(secondsPerYear).toNumber();
+  }
+
+  async getSystemWideDebtCeiling() {
+    const Line = await this._pitContract().Line();
+    return new BigNumber(Line.toString()).dividedBy(WAD).toNumber();
```

Comment:
Should we be using `toFixed` rather than `toString` on some of these values that could be very large? cc @levity

Reply:
Oh yes definitely. @tyler17 please test this with a nice aspirational number like $10 trillion :) you can use the setDebtCeiling test helper I added

Reply:
Sounds good, there's probably some functions in the SCD SDK that should be updated from `ToString` to `ToFixed` as well, since that's what I used for inspiration

Reply:
Actually, I'm not sure I understand the need to to do ToFixed, as it just gets converted to a BigNumber.  I tested the debt ceiling with 10T, and an even bigger number, and it didn't seem to be an issue

Reply:
I see what's happening here. Ethers uses a different bignumber library (`bn.js`) than we do elsewhere (`bignumber.js`). And the former never converts to scientific notation; its `toString` behavior is like the `toFixed` behavior of the latter.

Reply:
Ah got it.  So I'll just leave it as is then.